### PR TITLE
Fix OpticalFlow test premature exit on sm < 8

### PR DESCRIPTION
**Pull request by JanuszL**
- fixes premature exit of OpticalFlow test on sm < 8

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- fixes premature exit of OpticalFlow test on sm < 8

### Affected modules and functionalities:
- dali/test/python/test_optical_flow.py

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - dali/test/python/test_optical_flow.py
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/test/python/test_optical_flow.py [hunk 1]
```diff
@@ -234,38 +235,40 @@ def check_optflow(output_grid=1, hint_grid=1, use_temporal_hints=False):
                        hint_grid=hint_grid, use_temporal_hints=use_temporal_hints)
     pipe.build()
     if get_arch() < 8:
-        if output_grid != 4 and (hint_grid in [4, 8, None]):
+        if output_grid != 4:
             assert_raises(RuntimeError, pipe.run,
                           glob="grid size: * is not supported, supported are:")
+            raise SkipTest('Skipped as grid size is not supported for this arch')
+
         if output_grid == 4 and hint_grid not in [4, 8, None]:
```
Comment (mzient) [inline dali/test/python/test_optical_flow.py:243]:
```suggestion
        elif and hint_grid not in [4, 8, None]:
```

**Review by mzient — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Pull request merged**